argmin-py/src/executor.rs

```diff
@@ -0,0 +1,71 @@
+// Copyright 2018-2023 argmin developers
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://apache.org/licenses/LICENSE-2.0> or the MIT license <LICENSE-MIT or
+// http://opensource.org/licenses/MIT>, at your option. This file may not be
+// copied, modified, or distributed except according to those terms.
+
+// TODO: docs
+
+use pyo3::{prelude::*, types::PyDict};
+
+use argmin::core;
+
+use crate::problem::Problem;
+use crate::solver::{DynamicSolver, Solver};
+use crate::types::{IterState, PyArray1};
+
+#[pyclass]
+pub struct Executor(Option<core::Executor<Problem, DynamicSolver, IterState>>);
+
+impl Executor {
+    /// Consumes the inner executor.
+    ///
+    /// PyObjects do not allow methods that consume the object itself, so this is a workaround
+    /// for using methods like `configure` and `run`.
+    fn take(&mut self) -> anyhow::Result<core::Executor<Problem, DynamicSolver, IterState>> {
```

Comment:
Nice! I see the point for `run`, but I don't really see it for `configure`. In case of `run` the `Executor` is consumed, but in the python example, `configure` is used as if it were using `&mut self`. I am probably missing something.

Reply:
Thanks for the explanation!